avocado/utils/partition.py

```diff
@@ -230,10 +231,11 @@ def mount(self, mountpoint=None, fstype=None, args=''):
         args = args.lstrip()
 
         with MtabLock():
-            if self.device in self.list_mount_devices():
-                raise PartitionError(self, "Attempted to mount mounted device")
-            if mountpoint in self.list_mount_points():
-                raise PartitionError(self, "Attempted to mount busy directory")
+            if not bind:
+                if self.device in self.list_mount_devices():
+                    raise PartitionError(self, "Attempted to mount mounted device")
+                if mountpoint in self.list_mount_points():
+                    raise PartitionError(self, "Attempted to mount busy directory")
             if not os.path.isdir(mountpoint):
                 os.makedirs(mountpoint)
             try:
```

Comment:
Hi @harish-24 thanks for this.

But, Unless I'm missing something, this shouldn't also add a `-o bind` to the mount options? What I'm missing?

Reply:
@beraldoleal What you have specified is an option to mount using bind. This can be done through `args`. 
This patch is to allow mount to happen if a device or mountpoint is already mounted. 
Below is a usecase for the patch
```
/dev/sda1 on /mnt/1 type ext4 (rw,relatime,seclabel,dax)
/dev/sda1 on /1 type ext4 (rw,relatime,seclabel,dax)
```
or
```
/dev/sda1 on /mnt type ext4 (rw,relatime,seclabel,dax)
/dev/sda2 on /mnt type ext4 (rw,relatime,seclabel,dax)
```
Both must be allowed provided the kernel supports bind mounting. 
With current implementation there is no option to mount if a mountpoint or device is already in the mounted list. This is true even for devices like `tmpfs`, `none` too. This will allow the user to skip the check if needed.

Reply:
I see now. Thanks for the explanation.

In that case I would recommend to use a different name because a user (like myself) could think that using bind=True will result in `-o bind`.  Maybe something like `force_mount`, `mount_anyway` or something like that.

Because when a user reads "bind" he/she could think that this is related to `-o bind` option.

What do you think?

Reply:
yes that makes sense. will adapt to that.